cachito/workers/celery_logging.py

```diff
@@ -0,0 +1,111 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import inspect
+import logging
+
+from celery.utils.log import ColorFormatter
+
+
+class RequestFilter(logging.Filter):
+    """A log filter that sets ``request_id`` on the log record."""
+
+    def __init__(self, request_id, *args, **kwargs):
+        """
+        Initialize a filter that sets ``request_id`` on the log record.
+
+        :param request_id: the request ID to set on the record
+        """
+        super().__init__(*args, **kwargs)
+        self._request_id = request_id
+
+    def filter(self, record):
```

Comment:
We discussed the other day how we are using the filter method to set the request id, but should we rename it to something better?

Reply:
> You don’t need to create specialized Filter classes, or use other classes with a filter method: you can use a function (or other callable) as a filter. The filtering logic will check to see if the filter object has a filter attribute: if it does, it’s assumed to be a Filter and its filter() method is called. Otherwise, it’s assumed to be a callable and called with the record as the single parameter. The returned value should conform to that returned by filter().

From https://docs.python.org/3/library/logging.html#logging.Filter

A simple callable would be nice, but we'll have to use closures to store the value of the `request_id`. Something like this:

```python
def make_filter(request_id):
  def set_request_id(record):
    record.request_id = request_id
    return True
  return set_request_id
```

Then use it like:
```python
log_filter = make_filter(str(request_id) or "unknown")
```

It'd be tricky to remove it, but certainly doable.

The class approach seems pretty straightforward to me as well.

Reply:
@lcarva I'm going to name it `AddRequestIDFilter`. It makes it easier to know what it is by the name. Thanks for the suggestion.